kmk/hid.py

```diff
@@ -221,3 +222,110 @@ def hid_send(self, evt):
         return self.devices[reporting_device_const].send_report(
             evt[1 : HID_REPORT_SIZES[reporting_device_const] + 1]
         )
+
+
+class BLEHID(AbstractHID):
+    try:
+        # TODO FIXME This is wrapped in a try block to prevent crashes on M4 devices
+        from adafruit_ble import BLERadio
+        from adafruit_ble.advertising.standard import ProvideServicesAdvertisement
+        from adafruit_ble.services.standard.hid import HIDService
+        from storage import getmount
+
+        BLE_APPEARANCE_HID_KEYBOARD = const(961)
+        # Hardcoded in CPy
+        MAX_CONNECTIONS = const(2)
+
+        def post_init(self, ble_name=str(getmount('/').label), **kwargs):
```

Comment:
This inherently means if I have two halves of a keyboard running stock firmware and I haven't changed the filesystem labels on either, I'll have name collisions. While BT addresses based on MAC addresses/GUIDs in general and thus it's not the end of the world, it's still confusing to say the least. Is there another way we can handle this?

Reply:
The split guide recommends changing your names in the readme. More importantly though, only one half will advertise HID functionality, so it should not be an issue. At least I only see 1 device on my split keyboard.

Reply:
I actually can't move that to the top level. I get errors. Feel free to give it a shot.

Reply:
I guess for now the strategy can be "trust that people will follow the docs"; we can address this further if it becomes an issue.

Reply:
It's not a matter of "trust" as it is, if you fail to set either a name, or a split_side, it won't work, and never would work. This assumes that at least one is correct, and if none, the outcome is the same.